outline/shadowsocks/config.go

```diff
@@ -48,3 +79,37 @@ func NewClient(config *Config) (*Client, error) {
 
 	return &Client{c}, nil
 }
+
+// newConfig converts a UTF-8 JSON string into a Config struct.
+// A string is used to ensure that configs can be passed through gobind.
+func newConfig(s string) (*Config, error) {
+	var parsed configJSON
+	if err := json.Unmarshal([]byte(s), &parsed); err != nil {
+		return nil, err
+	}
+	if len(parsed.Host) == 0 || parsed.Port == 0 || len(parsed.Method) == 0 || len(parsed.Password) == 0 {
+		return nil, errors.New("missing mandatory field")
+	}
+	config := Config{
+		Host:       parsed.Host,
+		Port:       int(parsed.Port),
+		CipherName: parsed.Method,
+		Password:   parsed.Password,
+	}
+	prefixBytes, err := extractPrefixBytes(parsed.Prefix)
+	if err != nil {
+		return nil, fmt.Errorf("prefix parsing failed: %w", err)
+	}
+	config.Prefix = prefixBytes
+	return &config, nil
+}
+
+// MakeClient provides a gobind-compatible wrapper for [client.NewClient].
+// [configStr] contains a JSON configuration object.
+func MakeClient(configStr string) (*Client, error) {
```

Comment:
This function is not aligned with where we want to go. It returns a shadowsocks.Client. However we want it to be decoupled from Shadowsocks. This function will soon be deprecated.

What's your plan to address that?

Should we define a generic Client interface in this PR, so the new function can return that instead? In that case the function should probably move somewhere else.

Reply:
Incremental change is hard to combine with API stability.  In #108, I've resolved the conflict by moving this function into a new package.